python-package/lightgbm/basic.py

```diff
@@ -0,0 +1,1080 @@
+
+        Returns
+        -------
+        info : array
+            a numpy array of information of the data
+        """
+        tmp_out_len = ctypes.c_int64()
+        out_type = ctypes.c_int32()
+        ret = ctypes.POINTER(ctypes.c_void_p)()
+        _safe_call(_LIB.LGBM_DatasetGetField(
+            self.handle,
+            c_str(field_name),
+            ctypes.byref(tmp_out_len),
+            ctypes.byref(ret),
+            ctypes.byref(out_type)))
+        if out_type.value != FIELD_TYPE_MAPPER[field_name]:
+            raise TypeError("Return type error for get_field")
+        if tmp_out_len.value == 0:
+            return None
+        if out_type.value == C_API_DTYPE_INT32:
+            return cint32_array_to_numpy(ctypes.cast(ret, ctypes.POINTER(ctypes.c_int32)), tmp_out_len.value)
+        elif out_type.value == C_API_DTYPE_FLOAT32:
+            return cfloat32_array_to_numpy(ctypes.cast(ret, ctypes.POINTER(ctypes.c_float)), tmp_out_len.value)
+        else:
+            raise TypeError("unknow type")
+
+    def set_field(self, field_name, data):
+        """Set property into the Dataset.
+
+        Parameters
+        ----------
+        field_name: str
+            The field name of the information
+
+        data: numpy array or list or None
+            The array ofdata to be set
+        """
+        if data is None:
+            _safe_call(_LIB.LGBM_DatasetSetField(
+                self.handle,
+                c_str(field_name),
+                None,
+                0,
+                FIELD_TYPE_MAPPER[field_name]))
+            return
+        if not is_numpy_1d_array(data):
+            raise TypeError("Unknow type({})".format(type(data).__name__))
+        if data.dtype == np.float32:
+            ptr_data = data.ctypes.data_as(ctypes.POINTER(ctypes.c_float))
+            type_data = C_API_DTYPE_FLOAT32
+        elif data.dtype == np.int32:
+            ptr_data = data.ctypes.data_as(ctypes.POINTER(ctypes.c_int32))
+            type_data = C_API_DTYPE_INT32
+        else:
+            raise TypeError("excepted np.float32 or np.int32, met type({})".format(data.dtype))
+        if type_data != FIELD_TYPE_MAPPER[field_name]:
+            raise TypeError("type error for set_field")
+        _safe_call(_LIB.LGBM_DatasetSetField(
+            self.handle,
+            c_str(field_name),
+            ptr_data,
+            len(data),
+            type_data))
+
+
+    def save_binary(self, filename):
+        """Save Dataset to binary file
+
+        Parameters
+        ----------
+        filename : string
+            Name of the output file.
+        """
+        _safe_call(_LIB.LGBM_DatasetSaveBinary(
+            self.handle,
+            c_str(filename)))
+
+    def set_label(self, label):
+        """Set label of Dataset
+
+        Parameters
+        ----------
+        label: array like
+            The label information to be set into Dataset
+        """
+        label = list_to_1d_numpy(label, np.float32)
```

Comment:
I think the logic `if label.dtype != np.float32: label = label.astype(np.float32, copy=False)` can be put into `list_to_1d_numpy`. It occurs several times below.